### PR TITLE
Add pre-commit hook for custom_operator_name

### PR DESCRIPTION
**Pull request by uranusjr**
As mentioned in https://github.com/apache/airflow/pull/25783#pullrequestreview-1076765975

This helps `@task` operators to include a `custom_operator_name` attribute so they don't display an internal class name.

The check is not extensive (it ignores many special cases such as aliases), but all decorated
operators implemented following the contribution guide will be covered. And if a class does not follow the guide, it will need manual reviews to figure out a reasoning anyway.

### CODE REVIEW

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
nice!

File: airflow/decorators/branch_python.py [hunk 1]
```diff
@@ -38,6 +38,8 @@ class _BranchPythonDecoratedOperator(DecoratedOperator, BranchPythonOperator):
         Defaults to False.
     """
 
+    custom_operator_name = "@task.branch"
+
```
Comment (eladkal) [inline airflow/decorators/branch_python.py]: since https://github.com/apache/airflow/pull/25783 was merged no need to add it again

**Review by eladkal — APPROVED** [review]
cool!

File: airflow/decorators/branch_python.py (hunk 1 above)
Comment (potiuk) [reply airflow/decorators/branch_python.py]: It will be automatically "merged" on merge :)

**Review by josh-fell — APPROVED** [review]

File: airflow/decorators/branch_python.py (hunk 1 above)
Comment (uranusjr) [inline airflow/decorators/branch_python.py]:
```suggestion
```

**Pull request merged**